beacon_node/eth1/src/deposit_cache.rs

```diff
@@ -79,19 +80,51 @@ impl DepositDataTree {
     }
 }
 
+#[derive(Encode, Decode, Clone)]
+pub struct SszDepositCache {
+    logs: Vec<DepositLog>,
+    leaves: Vec<Hash256>,
+    deposit_contract_deploy_block: u64,
+    deposit_roots: Vec<Hash256>,
+}
+
+impl SszDepositCache {
+    pub fn from_deposit_cache(cache: &DepositCache) -> Self {
+        Self {
+            logs: cache.logs.clone(),
+            leaves: cache.leaves.clone(),
+            deposit_contract_deploy_block: cache.deposit_contract_deploy_block,
+            deposit_roots: cache.deposit_roots.clone(),
+        }
+    }
+
+    pub fn to_deposit_cache(&self) -> Result<DepositCache, String> {
+        let deposit_tree =
```

Comment:
IIRC, building this `DepositDataTree` is quite computationally expense (30s?). Can we ser/deser this too?

Reply:
Ser/Deser the `DepositDataTree` boils down to encoding/decoding the `MerkleTree`. Encoding/decoding the `MerkleTree` isn't very straightforward. Currently, it takes around a second to create a tree of a million deposits, which is good enough for testnet. So it might be better to move this to an issue of its own imo. 
Let me know what you think :)

Reply:
Opened an issue for this #776